### DISCUSSION TITLE
Apollo queries on server from react tree

### DISCUSSION THREAD

**Discussion by SeeThruHead** [vercel/next.js#21664] [Help]
So we have an interesting usecase maybe not seen that often.

We have a configuration api that we hit in getInitialProps, it returns us a data structure. We use this data structure to build our react tree. That react tree contains many different calls to useQuery.

Since we are rendering the react tree on the server-side. Shouldn't we be able to extract all the queries from the tree? Maybe out of context?

Currently you have to manually query from getInitialProps. But that means manually exporting all our queries and calling them directly.

I would be more interesting in rendering the react tree and extracting the queries from it inside getInitialProps.
Has anyone done anything like this? Or have some recommendations?

**Comment by lifeiscontent** [comment]
I also wonder this, AFAIK it's not possible to use the apollo `getDataFromTree` if any of your queries depend on query params because the next router context is not set yet for the root component you're trying to hydrate from getServerSideProps.
